languages/tree-sitter-stack-graphs-typescript/src/stack-graphs.tsg

```diff
@@ -541,7 +541,9 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
   ; module reference
   var mod_scope = mod_ref__ns
   scan (path-normalize mod_path) {
-    "([^/]+)/?" {
+    ; ignore .js and .ts extensions on imports, as ts allows this
+    ; this might lead to false positives
+    "([^/|(\.j|ts)]+)/?" {
```

Comment:
I think the idea is correct, bit this doesn't look quite right. What about this?

```suggestion
    "([^/]+)(/|\.ts|\.tsx|\.js|\.jsx\)?" {
```

Reply:
Actually my regex is a bit of a hack but as .js (and .ts, ...) are included in this broad pattern: `([^/]+)` it will be included if this is optional: `(/|\.ts|\.tsx|\.js|\.jsx\)` (eg. it will still match the extention). Working on something cleaner.

Reply:
It seems the regex parser does not support non greedy matching ? `(...?)`

Reply:
Ah, I see the problem! An alternative might be to keep the original regex, and strip any of the extensions off inside the body using `(replace ...)`. That might be easier.

Reply:
Oh sorry posted without refreshing, missed your comment. What do you prefer between replace and the new util function ?

Reply:
I think you might be able to write this with existing functions already. Something like this:

    (path-normalize (path-join (path-dir mod_path) (path-stem mod_path)))

Let's do that instead of adding custom code.

Reply:
You are right, i'll edit the PR by the end of the week / weekend.

Reply:
So it turns out that `path-stem` is actually `path-filestem`, and it can return null (eg on directories (`"."`)). 
My intuition was to check if it's null then use `""`, but i'm having a hard time doing it, because of what I pointed out in this issue: https://github.com/tree-sitter/tree-sitter-graph/issues/154.

Reply:
Also, path-dir ".." will return "", which breaks some cases (`import {} from '..'` in `test/modules/import-reexports-from-superdirectory-index.ts`)

Reply:
I also tried a way using regexes and replace, but it seems the regex in tsg does not strictly behave as in rust:
Given the regex `\.[^\.\\/]+$`
```
scan "./bar" {
    "\.[^\.\\/]+$" {
      print $0
    }
 }
```
Outputs `/bar`
But using the same regex in rust it does not match (as expected):
https://regex101.com/r/Rwm8lq/1 (check the unit tests tab)

Reply:
I believe the implementation of `path-dir` is kind of wrong:
```rust
functions.add(
    "path-dir".into(),
    path_fn(|p| p.parent().map(|s| s.as_os_str().to_os_string())),
);
```
For instance the dir of `.` is `.` and the dir of `..` is `..` ?

Reply:
I got it to work by removing the explicit extensions (.js, .ts, .jsx, .tsx, not all extensions) with: `(path-normalize (replace mod_path "\.(j|t)sx?$" ""))`, but the questions above still remain and should be considered I believe

Reply:
(tho this can be merged I believe)

Reply:
My kingdom for a small coherent theory of paths 😭. Thanks for figuring out these cases. I'll put them in an issue so we document this in a more visible place.